File: src/main/kotlin/com/screentech/cordamigrate/entity/passwordreset/PasswordReset.java
package com.screentech.cordamigrate.entity.passwordreset;

import javax.persistence.*;
import java.math.BigDecimal;
import java.sql.Date;

@Entity
@Table(name = "password_resets")
public class PasswordReset {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private BigDecimal id;

    @Column(name = "email")
    private String email;

    @Column(name = "token")
    private String token;

    @Column(name = "create_at", updatable = false, insertable = false)
    private Date createAt;

    public PasswordReset() {
    }

    public BigDecimal getId() {
        return id;
    }

    public void setId(BigDecimal id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public Date getCreateAt() {
        return createAt;
    }

    public void setCreateAt(Date createAt) {
        this.createAt = createAt;
    }

    @Override
    public String toString() {
        return "PasswordReset{" +
                "id=" + id +
                ", email='" + email + '\'' +
                ", token='" + token + '\'' +
                ", createAt=" + createAt +
                '}';
    }
}

File: src/main/kotlin/com/screentech/cordamigrate/entity/passwordreset/PasswordReset.kt
package com.screentech.cordamigrate.entity.passwordreset

import com.screentech.cordamigrate.utility.getCurrentTimestampSQL
import java.sql.Timestamp
import javax.persistence.*

@Entity
@Table(name = "password_resets")
class PasswordReset (
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY) @Column(name = "id") var id : Long,
        @Column(name = "email", columnDefinition = "varchar(255) not null") var email: String? = null,
        @Column(name = "token", columnDefinition = "text") var token: String? = null,
        @Column(name = "create_at", columnDefinition = "timestamp default now()") var createAt: Timestamp? = getCurrentTimestampSQL(),
        @Transient var createAtStr: String?
)
